Small tidy-ups of TensorMechanics

WilkAndy opened the pull request:
Changed initialise to initialize (spelling) in a number of TensorMechanics classes
ElasticityTensorTools: added getIsotropicPoissonsRatio
CappedMohrCoulombStressUpdate: used getIsotropicPoissonsRatio
tests/combined_creep_plasticity/tests: changed super_lu=true to superlu=true.
MultiParameterPlasticityStressUpdate and derived classes: consistentTangentOperator function is only called if currentlyComputingJacobian() = true

Fixes #9602

WilkAndy committed:
Changed initialise to initialize (spelling) in a number of TensorMechanics classes
ElasticityTensorTools: added getIsotropicPoissonsRatio
CappedMohrCoulombStressUpdate: used getIsotropicPoissonsRatio
tests/combined_creep_plasticity/tests: changed super_lu=true to superlu=true.
MultiParameterPlasticityStressUpdate and derived classes: consistentTangentOperator function is only called if currentlyComputingJacobian() = true

Fixes #9602

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/105290/) on 72ac57b wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/9611/site)

WilkAndy committed:
Tests use TensorMechanics/Master Action.

Some minor re-golding due to me wanting to use infinitesimal strains rather than finite strains.

Fixes #9602

permcody reviewed (approved):
Nice line reduction!

WilkAndy commented:
Thanks for the review and merge, @permcody ,

a
